datafusion/sqllogictest/README.md

```diff
@@ -102,6 +102,39 @@ SELECT * from foo;
 
 Assuming it looks good, check it in!
 
+## Cookbook: Testing for whitespace
+
+The `sqllogictest` runner will automatically strip trailing whitespace, meaning
+it additional effort to verify that trailing whitespace is correctly produced
+
+For example, the following test can't distinguish between `Andrew` and `Andrew `
+(with trailing space):
+
+```text
+query T
+select substr('Andrew Lamb', 1, 7)
+----
+Andrew
+```
+
+To test trailing whitespace, project additional non-whitespace column on the
+right. For example, by selecting `'XX'` after the column of interest, the test
```

Comment:
It's any non-whitespace, but we can take opportunity to suggest what we want to see in tests.
What about `'|'` that's used in one test i added, or `'$'` (not non-typical for marking end of line)?

Reply:
Let's use `|` to be consistent with what you added . I will update the PR